decorators.go

```diff
@@ -242,6 +248,71 @@ func (CommandWithFlagsDecorator) Decorate(_ *Ecdysis, cmd *cobra.Command, c Comm
 	return nil
 }
 
+// -- PARSING CONFIGURATION --------------------------------------------------------------------
+
+// CommandWithConfiguration can be implemented by a command to parsing configuration.
+type CommandWithConfiguration interface {
+	Command
+
+	ParseConfig() Config
+}
+
+// CommandWithParsingConfigDecorator is a decorator that sets the command flags.
+type CommandWithParsingConfigDecorator struct{}
+
+// Decorate parses the configuration based on flags.
+func (CommandWithParsingConfigDecorator) Decorate(_ *Ecdysis, cmd *cobra.Command, c Command) error {
+	v, ok := c.(CommandWithConfiguration)
+	if !ok {
+		return nil
+	}
+
+	old := cmd.PreRunE
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
+		if old != nil {
+			err := old(cmd, args)
+			if err != nil {
+				return err
+			}
+		}
+
+		usrCfg := v.ParseConfig()
+
+		// Ensure ParsedCfg is a pointer
+		if reflect.ValueOf(usrCfg.ParsedCfg).Kind() != reflect.Ptr {
+			return fmt.Errorf("ParsedCfg must be a pointer")
+		}
+
+		viper := viper.New()
+
+		// Set default values
+		setDefaults(viper, usrCfg.DefaultCfg)
+
+		// Handle env variables
+		viper.SetEnvPrefix(usrCfg.EnvPrefix)
+		viper.AutomaticEnv()
+		viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+
+		// Handle config file
+		viper.SetConfigFile(usrCfg.ConfigPath)
+		if err := viper.ReadInConfig(); err != nil {
+			return fmt.Errorf("fatal error config file: %w", err)
+		}
+
+		// Handle flags
+		cmd.Flags().VisitAll(func(f *pflag.Flag) {
+			if err := viper.BindPFlag(f.Name, f); err != nil {
+				fmt.Printf("error binding flag: %v\n", err)
```

Comment:
Shouldn't we make Decorate() method return an error when this happens?

Reply:
https://github.com/ConduitIO/ecdysis/pull/3/commits/03472b66d26a5675aa4eeb02fddf6719e53efdbb